### PR TITLE
feat(node/process): `process.getuid()` and `process.getgid()`

### PR DESCRIPTION
**Pull request by iuioiua**
1. Is this the best way to conditionally define a method of a class? These methods don't exist on Windows platforms in Node.
2. How can `// @ts-ignore` be avoided in the tests?
3. [`test-process-uid-gid.js`](https://github.com/nodejs/node/blob/main/test/parallel/test-process-uid-gid.js) wasn't added because `process.setuid()`, `process.seteuid()`, `process.setgid()` and `process.setegid()` aren't and cannot currently be implemented, meaning the majority of that test would fail.

### CODE REVIEW

File: node/process.ts [hunk 1]
```diff
@@ -560,18 +560,6 @@ class Process extends EventEmitter {
     return 0o22;
   }
 
-  /** https://nodejs.org/api/process.html#processgetuid */
-  getuid(): number {
```
Comment (kt3k) [inline node/process.ts:564]: how about keeping only the typing like `getuid?: () => number;` and call it like `getuid?.()` in testing? (I guess that reduces the type errors in test cases)

Note: DefinitelyTyped defines this as `getuid?: () => number;` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/35b81ce6b7865843f9c2e49861c470621134ce3b/types/node/process.d.ts#L689

**Comment by kt3k** [comment]
@iuioiua Thanks for looking into this! Left one comment

**Comment by cjihrig** [comment]
I have to ask, are there any Node.js tests that can be ported over with these APIs?

**Comment by iuioiua** [comment]
> I have to ask, are there any Node.js tests that can be ported over with these APIs?

See no. 3 in my opening comment.

**Review by cjihrig — APPROVED** [review]

**Pull request merged**